### PR TITLE
feat!: passing operartor address to the RDK

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

This PR implmements `dymint` side of the new sequencer registration flow.
The RDK expects dymint to pass 2 pubkeys on the `InitChain` request:
1. The consensus pubkey
2. The operator address pubkey

To get the operator addr, we will have dedicated config in `dymint.toml`
which will be defaulted to the sequencer key used for state updates

**Additional changes:**
* Init chain is called in aggregator mode only
* local priv key is checked against the Proposer registered on the hub
* go version updated

---
For Author:

- [x]  Targeted PR against correct branch
- [x]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/manager.go [hunk 1]
```diff
@@ -315,49 +298,4 @@ func (m *Manager) getLatestBatchFromSL(ctx context.Context) (*settlement.ResultR
 		return resultRetrieveBatch, err
 	}
 	return resultRetrieveBatch, nil
-
-}
-
-// TODO(omritoptix): possible remove this method from the manager
-func updateInitChainState(s *types.State, res *abci.ResponseInitChain, validators []*tmtypes.Validator) {
```
Comment (mtsitrin) [inline block/manager.go:322]: moved to executer

**Comment by danwt** [comment]
Is this ready for review @mtsitrin ? Linter and cov is failing

**Comment by mtsitrin** [comment]
> Is this ready for review @mtsitrin ? Linter and cov is failing

linter seems to fail on some unknown env reason. PR ready to review

**Comment by danwt** [comment]
connected to https://github.com/dymensionxyz/dymension-rdk/commit/07ca2a958a619179a2a3a879e2b82fc4fa394d9e

File: block/initchain.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/evmos/evmos/v12/crypto/hd"
+	"github.com/ignite/cli/ignite/pkg/cosmosaccount"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+func (m *Manager) RunInitChain(ctx context.Context) error {
+	//get the proposer's consensus pubkey
```
Comment (danwt) [inline block/initchain.go:19]:
```suggestion
func (m *Manager) RunInitChain(ctx context.Context) error {

	// We pass an initial val set which is actually the consensus and operator addresses of the proposer.
	// This is a hack to make sure the chain can get both addresses without us needing to change comet signatures:
	//  The RDK will save a sequencer, and delete the extra validator after initChain, so we also delete it here
	//  to keep the state in sync.
	
	//get the proposer's consensus pubkey
```

File: block/manager.go [hunk 3]
```diff
@@ -128,29 +128,6 @@ func NewManager(
...
-		if err != nil {
-			return nil, err
-		}
-
-		updateInitChainState(&s, res, validators)
-		if _, err := store.UpdateState(s, nil); err != nil {
-			return nil, err
-		}
-	}
-
```
Comment (danwt) [inline block/manager.go:153]: I guess moving this to Start is just a refactor?

**Review by danwt — DISMISSED** [review]
Looks good

I suggested an explanation comment

One question about terminology:

Does aggregator mode mean that the node is the current active sequencer (= proposer)?

**Comment by danwt** [comment]
Is the linter thing something to do with mocks?

File: block/manager.go (hunk 3 above)
Comment (mtsitrin) [reply block/manager.go:153]: also called now only by the aggregator (was in common flow before)

Comment (danwt) [reply block/manager.go:153]: Ah ok, in terminology, does aggregator = current active sequencer = proposer?

Comment (mtsitrin) [reply block/manager.go:153]: terminology wise, yes.

initially aggreagator mode was only set by a flag in dymint.toml,
in this PR I've added validation to this assumption
(local key matches the proposer sequencer)

**Review by danwt — APPROVED** [review]

**Review by srene — APPROVED** [review]

**Pull request merged**